### ISSUE TITLE
Implement a re-try for iOS 9 simulator - UI tests

### ISSUE THREAD

**Issue by viktorbenei** [bitrise-io/steps-xcode-test#7]
Similar to the existing "boot timeout" issue.

iOS 9 simulator sometimes fail to prepare for UI tests if the simulator is not yet booted.

With our Xcode 7 sample the logs look like this:
```
Touch /Users/vagrant/Library/Developer/Xcode/DerivedData/BitriseXcode7Sample-adkzkiqwlfliqgcggdmlvbmsomkl/Build/Products/Debug-iphonesimulator/BitriseXcode7SampleUITests-Runner.app/PlugIns/BitriseXcode7SampleUITests.xctest
    cd /Users/vagrant/git
    export PATH="/Applications/Xcodes/Xcode-beta.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcodes/Xcode-beta.app/Contents/Developer/usr/bin:/Users/vagrant/.rvm/gems/ruby-2.1.5/bin:/Users/vagrant/.rvm/gems/ruby-2.1.5@global/bin:/Users/vagrant/.rvm/rubies/ruby-2.1.5/bin:/usr/local/bin:/usr/local/sbin:~/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/opt/go/libexec/bin:/Users/vagrant/go/bin:/Users/vagrant/bitrise/tools/cmd-bridge/bin/osx:/Users/vagrant/.rvm/bin:/Users/vagrant/.rvm/bin"
    /usr/bin/touch -c /Users/vagrant/Library/Developer/Xcode/DerivedData/BitriseXcode7Sample-adkzkiqwlfliqgcggdmlvbmsomkl/Build/Products/Debug-iphonesimulator/BitriseXcode7SampleUITests-Runner.app/PlugIns/BitriseXcode7SampleUITests.xctest
2015-09-17 04:11:40.366 xcodebuild[924:5258] [MT] IDETestOperationsObserverDebug: (C96AB9F2-D4B7-43D1-BDDB-793AA2E0C473) Beginning test session with Xcode 7A192o on target  {
		SimDevice: SimDevice : iPhone 6 (058C71C4-46CB-4878-930E-6A0E8707B917) : state={ Booted } deviceType={ SimDeviceType : com.apple.CoreSimulator.SimDeviceType.iPhone-6 } runtime={ SimRuntime : 9.0 (13A4325c) - com.apple.CoreSimulator.SimRuntime.iOS-9-0 }
} (9.0 (13A4325c))
** TEST FAILED **
Test Suite 'All tests' started at 2015-09-17 04:11:25.924
Test Suite 'BitriseXcode7SampleTests.xctest' started at 2015-09-17 04:11:25.927
Test Suite 'BitriseXcode7SampleTests' started at 2015-09-17 04:11:25.945
Test Case '-[BitriseXcode7SampleTests.BitriseXcode7SampleTests testExample]' started.
Test Case '-[BitriseXcode7SampleTests.BitriseXcode7SampleTests testExample]' passed (0.004 seconds).
Test Case '-[BitriseXcode7SampleTests.BitriseXcode7SampleTests testPerformanceExample]' started.
/Users/vagrant/git/BitriseXcode7SampleTests/BitriseXcode7SampleTests.swift:33: Test Case '-[BitriseXcode7SampleTests.BitriseXcode7SampleTests testPerformanceExample]' measured [Time, seconds] average: 0.000, relative standard deviation: 197.924%, values: [0.000012, 0.000001, 0.000001, 0.000000, 0.000000, 0.000001, 0.000001, 0.000001, 0.000001, 0.000000], performanceMetricID:com.apple.XCTPerformanceMetric_WallClockTime, baselineName: "", baselineAverage: , maxPercentRegression: 10.000%, maxPercentRelativeStandardDeviation: 10.000%, maxRegression: 0.100, maxStandardDeviation: 0.100
Test Case '-[BitriseXcode7SampleTests.BitriseXcode7SampleTests testPerformanceExample]' passed (0.476 seconds).
Test Suite 'BitriseXcode7SampleTests' passed at 2015-09-17 04:11:26.433.
	 Executed 2 tests, with 0 failures (0 unexpected) in 0.480 (0.488) seconds
Test Suite 'BitriseXcode7SampleTests.xctest' passed at 2015-09-17 04:11:26.434.
	 Executed 2 tests, with 0 failures (0 unexpected) in 0.480 (0.507) seconds
Test Suite 'All tests' passed at 2015-09-17 04:11:26.436.
	 Executed 2 tests, with 0 failures (0 unexpected) in 0.480 (0.512) seconds
2015-09-17 04:11:28.826 XCTRunner[1074:6537] Running tests...
2015-09-17 04:11:28.828 XCTRunner[1074:6537] Looking for test bundles in /Users/vagrant/Library/Developer/CoreSimulator/Devices/058C71C4-46CB-4878-930E-6A0E8707B917/data/Containers/Bundle/Application/05469575-8790-4EFD-B7D5-00F4A983F5B1/BitriseXcode7SampleUITests-Runner.app/PlugIns
2015-09-17 04:11:28.832 XCTRunner[1074:6537] Found test bundle at /Users/vagrant/Library/Developer/CoreSimulator/Devices/058C71C4-46CB-4878-930E-6A0E8707B917/data/Containers/Bundle/Application/05469575-8790-4EFD-B7D5-00F4A983F5B1/BitriseXcode7SampleUITests-Runner.app/PlugIns/BitriseXcode7SampleUITests.xctest
2015-09-17 04:11:28.832 XCTRunner[1074:6537] Looking for configurations in /Users/vagrant/Library/Developer/CoreSimulator/Devices/058C71C4-46CB-4878-930E-6A0E8707B917/data/Containers/Bundle/Application/05469575-8790-4EFD-B7D5-00F4A983F5B1/BitriseXcode7SampleUITests-Runner.app/PlugIns/BitriseXcode7SampleUITests.xctest
2015-09-17 04:11:28.835 XCTRunner[1074:6537] Found configuration 
	                  testBundleURL:file:///Users/vagrant/Library/Developer/Xcode/DerivedData/BitriseXcode7Sample-adkzkiqwlfliqgcggdmlvbmsomkl/Build/Products/Debug-iphonesimulator/BitriseXcode7SampleUITests-Runner.app/PlugIns/BitriseXcode7SampleUITests.xctest
	              productModuleName:BitriseXcode7SampleUITests
	                    testsToSkip:(null)
	                     testsToRun:(null)
	             reportResultsToIDE:YES
	              sessionIdentifier:<__NSConcreteUUID 0x7ff562d98170> 854AAD31-BAE6-451E-935B-1B71B15170C5
	     pathToXcodeReportingSocket:(null)
	      disablePerformanceMetrics:no
	treatMissingBaselinesAsFailures:no
	                baselineFileURL:(null)
	          targetApplicationPath:/Users/vagrant/Library/Developer/Xcode/DerivedData/BitriseXcode7Sample-adkzkiqwlfliqgcggdmlvbmsomkl/Build/Products/Debug-iphonesimulator/BitriseXcode7Sample.app
	      targetApplicationBundleID:com.bitrise.BitriseXcode7Sample
	               reportActivities:no
Test Suite 'All tests' started at 2015-09-17 04:11:29.075
Test Suite 'BitriseXcode7SampleUITests.xctest' started at 2015-09-17 04:11:29.077
Test Suite 'BitriseXcode7SampleUITests' started at 2015-09-17 04:11:29.078
Test Case '-[BitriseXcode7SampleUITests.BitriseXcode7SampleUITests testAddAnItemGoToDetailsThenDeleteIt]' started.
    t =     0.00s     Start Test
    t =     0.00s     Set Up
    t =     0.02s         Launch com.bitrise.BitriseXcode7Sample
    t =     2.58s             Assertion Failure: UI Testing Failure - Failed to receive completion for 
/Users/vagrant/git/BitriseXcode7SampleUITests/BitriseXcode7SampleUITests.swift:21: error: -[BitriseXcode7SampleUITests.BitriseXcode7SampleUITests testAddAnItemGoToDetailsThenDeleteIt] : UI Testing Failure - Failed to receive completion for 
2015-09-17 04:11:31.709 XCTRunner[1074:6537] *** Terminating app due to uncaught exception '_XCTestCaseInterruptionException', reason: 'Interrupting test'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000010f9069b5 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x000000010f37edeb objc_exception_throw + 48
	2   CoreFoundation                      0x000000010f9068ed +[NSException raise:format:] + 205
	3   XCTest                              0x000000010ee418b9 -[XCTestCase _dequeueFailures] + 560
	4   XCTest                              0x000000010ee41b95 -[XCTestCase _enqueueFailureWithDescription:inFile:atLine:expected:] + 576
	5   XCTest                              0x000000010ee49340 _XCTFailureHandler + 1112
	6   XCTest                              0x000000010ee40d39 _XCTFailInCurrentTest + 512
	7   XCTest                              0x000000010ee5e663 -[XCUIDevice _dispatchEventWithPage:usage:duration:] + 790
	8   XCTest                              0x000000010ee4f7f8 -[XCAXClient_iOS init] + 153
	9   XCTest                              0x000000010ee4f756 __30+[XCAXClient_iOS sharedClient]_block_invoke + 24
	10  libdispatch.dylib                   0x0000000111b404c7 _dispatch_client_callout + 8
	11  libdispatch.dylib                   0x0000000111b2dd2f dispatch_once_f + 76
	12  XCTest                              0x000000010ee4f73c +[XCAXClient_iOS sharedClient] + 42
	13  XCTest                              0x000000010ee5fbe6 __37-[XCUIApplication _launchUsingXcode:]_block_invoke + 50
	14  XCTest                              0x000000010ee45443 -[XCTestCase startActivityWithTitle:block:] + 305
	15  XCTest                              0x000000010ee5fba5 -[XCUIApplication _launchUsingXcode:] + 309
	16  BitriseXcode7SampleUITests          0x000000011b0eedd9 _TFC26BitriseXcode7SampleUITests26BitriseXcode7SampleUITests5setUpfS0_FT_T_ + 153
	17  BitriseXcode7SampleUITests          0x000000011b0eee62 _TToFC26BitriseXcode7SampleUITests26BitriseXcode7SampleUITests5setUpfS0_FT_T_ + 34
	18  XCTest                              0x000000010ee45443 -[XCTestCase startActivityWithTitle:block:] + 305
	19  XCTest                              0x000000010ee41def __24-[XCTestCase invokeTest]_block_invoke_2 + 118
	20  XCTest                              0x000000010ee714a0 -[XCTestContext performInScope:] + 184
	21  XCTest                              0x000000010ee41d68 -[XCTestCase invokeTest] + 169
	22  XCTest                              0x000000010ee42203 -[XCTestCase performTest:] + 443
	23  XCTest                              0x000000010ee3fec9 -[XCTestSuite performTest:] + 377
	24  XCTest                              0x000000010ee3fec9 -[XCTestSuite performTest:] + 377
	25  XCTest                              0x000000010ee3fec9 -[XCTestSuite performTest:] + 377
	26  XCTest                              0x000000010ee2d6d2 __25-[XCTestDriver _runSuite]_block_invoke + 51
	27  XCTest                              0x000000010ee4d8db -[XCTestObservationCenter _observeTestExecutionForBlock:] + 615
	28  XCTest                              0x000000010ee2d61e -[XCTestDriver _runSuite] + 408
	29  XCTest                              0x000000010ee2e37d -[XCTestDriver _checkForTestManager] + 263
	30  XCTest                              0x000000010ee72801 _XCTestMain + 628
	31  libdispatch.dylib                   0x0000000111b24ea9 _dispatch_call_block_and_release + 12
	32  libdispatch.dylib                   0x0000000111b404c7 _dispatch_client_callout + 8
	33  libdispatch.dylib                   0x0000000111b2b07d _dispatch_main_queue_callback_4CF + 714
	34  CoreFoundation                      0x000000010f866e69 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
	35  CoreFoundation                      0x000000010f8283b9 __CFRunLoopRun + 2073
	36  CoreFoundation                      0x000000010f827918 CFRunLoopRunSpecific + 488
	37  GraphicsServices                    0x000000011136aad2 GSEventRunModal + 161
	38  UIKit                               0x000000010fcb499e UIApplicationMain + 171
	39  XCTRunner                           0x000000010edb86bf XCTRunner + 5823
	40  libdyld.dylib                       0x0000000111b7092d start + 1
	41  ???                                 0x0000000000000005 0x0 + 5
)
libc++abi.dylib: terminating with uncaught exception of type _XCTestCaseInterruptionException
2015-09-17 04:11:42.871 XCTRunner[1079:6930] Running tests...
2015-09-17 04:11:42.876 XCTRunner[1079:6930] Looking for test bundles in /Users/vagrant/Library/Developer/CoreSimulator/Devices/058C71C4-46CB-4878-930E-6A0E8707B917/data/Containers/Bundle/Application/7326BF2E-AA95-4EBC-8564-20FBE0B7B18B/BitriseXcode7SampleUITests-Runner.app/PlugIns
2015-09-17 04:11:42.885 XCTRunner[1079:6930] Found test bundle at /Users/vagrant/Library/Developer/CoreSimulator/Devices/058C71C4-46CB-4878-930E-6A0E8707B917/data/Containers/Bundle/Application/7326BF2E-AA95-4EBC-8564-20FBE0B7B18B/BitriseXcode7SampleUITests-Runner.app/PlugIns/BitriseXcode7SampleUITests.xctest
2015-09-17 04:11:42.888 XCTRunner[1079:6930] Looking for configurations in /Users/vagrant/Library/Developer/CoreSimulator/Devices/058C71C4-46CB-4878-930E-6A0E8707B917/data/Containers/Bundle/Application/7326BF2E-AA95-4EBC-8564-20FBE0B7B18B/BitriseXcode7SampleUITests-Runner.app/PlugIns/BitriseXcode7SampleUITests.xctest
2015-09-17 04:11:42.895 XCTRunner[1079:6930] Found configuration 
	                  testBundleURL:file:///Users/vagrant/Library/Developer/Xcode/DerivedData/BitriseXcode7Sample-adkzkiqwlfliqgcggdmlvbmsomkl/Build/Products/Debug-iphonesimulator/BitriseXcode7SampleUITests-Runner.app/PlugIns/BitriseXcode7SampleUITests.xctest/
	              productModuleName:BitriseXcode7SampleUITests
	                    testsToSkip:BitriseXcode7SampleUITests/testAddAnItemGoToDetailsThenDeleteIt()
	                     testsToRun:(null)
	             reportResultsToIDE:YES
	              sessionIdentifier:<__NSConcreteUUID 0x7f8679d36f00> C96AB9F2-D4B7-43D1-BDDB-793AA2E0C473
	     pathToXcodeReportingSocket:(null)
	      disablePerformanceMetrics:no
	treatMissingBaselinesAsFailures:no
	                baselineFileURL:(null)
	          targetApplicationPath:/Users/vagrant/Library/Developer/Xcode/DerivedData/BitriseXcode7Sample-adkzkiqwlfliqgcggdmlvbmsomkl/Build/Products/Debug-iphonesimulator/BitriseXcode7Sample.app
	      targetApplicationBundleID:com.bitrise.BitriseXcode7Sample
	               reportActivities:no
Test Suite 'Selected tests' started at 2015-09-17 04:11:43.135
Test Suite 'BitriseXcode7SampleUITests.xctest' started at 2015-09-17 04:11:43.140
Test Suite 'BitriseXcode7SampleUITests.xctest' passed at 2015-09-17 04:11:43.144.
	 Executed 0 tests, with 0 failures (0 unexpected) in 0.000 (0.004) seconds
Test Suite 'Selected tests' passed at 2015-09-17 04:11:43.152.
	 Executed 0 tests, with 0 failures (0 unexpected) in 0.000 (0.017) seconds
2015/09/17 04:11:43 Failed to run xcode test, error: exit status 65
exit status 1
ERRO[04:11:43] Step (Xcode: Unit Test) failed, error: (exit status 1)
```

**Comment by viktorbenei** [comment]
probably the search string should be `terminating with uncaught exception of type _XCTestCaseInterruptionException` or `Assertion Failure: UI Testing Failure - Failed to receive completion for` maybe combined with `Terminating app due to uncaught exception` - but this have to be tested!

**Comment by viktorbenei** [comment]
This seems to be a known issue, and related to performance, as described https://forums.developer.apple.com/thread/15209 at the end

**Comment by viktorbenei** [comment]
Fixed in `1.2.0` - PR https://github.com/bitrise-io/steps-xcode-test/pull/9
